tests/components/test_wake_on_lan.py

```diff
@@ -0,0 +1,53 @@
+"""Tests for Wake On LAN component."""
+import asyncio
+from functools import partial
+from unittest import mock
+
+import pytest
+
+from homeassistant.setup import async_setup_component
+from homeassistant.components.wake_on_lan import (
+    DOMAIN, SERVICE_SEND_MAGIC_PACKET)
+
+
+@pytest.fixture
+def mock_wakeonlan():
+    """Mock mock_wakeonlan."""
+    module = mock.MagicMock()
+    with mock.patch.dict('sys.modules', {
+        'wakeonlan': module,
+    }):
+        yield module
+
+
+@asyncio.coroutine
+def test_setup_component(hass, mock_wakeonlan):
+    """Test the set up of new component."""
+    assert(not hass.services.has_service(DOMAIN, SERVICE_SEND_MAGIC_PACKET))
+    yield from async_setup_component(hass, DOMAIN, {})
+    assert(hass.services.has_service(DOMAIN, SERVICE_SEND_MAGIC_PACKET))
+
+
+@asyncio.coroutine
+def test_send_magic_packet(hass, caplog, mock_wakeonlan):
+    """Test of send magic packet service call."""
+    yield from async_setup_component(hass, DOMAIN, {})
+
+    yield from hass.async_add_job(partial(
```

Comment:
I can confirm that, when doing any of these:
```python
yield from hass.services.call(DOMAIN, SERVICE_SEND_MAGIC_PACKET, {"mac": mac}, blocking=True)
yield from hass.services.call(DOMAIN, SERVICE_SEND_MAGIC_PACKET, {"mac": mac}
hass.services.call(DOMAIN, SERVICE_SEND_MAGIC_PACKET, {"mac": mac}, blocking=True)
hass.services.call(DOMAIN, SERVICE_SEND_MAGIC_PACKET, {"mac": mac})
```
The test never ends, as if never starts ¿?

But when doing any of these, all things happen as expected:
```python
yield from hass.async_add_job(
    hass.services.call, DOMAIN, SERVICE_SEND_MAGIC_PACKET, {"mac": mac})
yield from hass.async_add_job(partial(
    hass.services.call, DOMAIN, SERVICE_SEND_MAGIC_PACKET, {"mac": mac}, blocking=True))
```

This is happening when `send_magic_packet(call)` is a callback, a coroutine, or nothing at all.

So I'm going to commit changes to set `send_magic_packet` as coroutine, doing `yield from hass.async_add_job` for the wol call and for the service calls in the test, and wait for review, ok?